dask_expr/_collection.py

```diff
@@ -1254,3 +1255,68 @@ def concat(
             *[df.expr for df in dfs],
         )
     )
+
+
+def from_map(
+    func,
+    *iterables,
+    args=None,
+    meta=no_default,
+    divisions=None,
+    label=None,
+    enforce_metadata=False,
+    allow_projection=True,
+    **kwargs,
+):
+    """Create a dask-expr collection from a custom function map"""
+    from dask_expr.io import FromMap, FromMapProjectable
+
+    if "token" in kwargs:
+        # This option doesn't really make sense in dask-expr
+        raise NotImplementedError("dask_expr does not support a token argument.")
+
+    if allow_projection:
```

Comment:
I might be missing something, but is it possible that we set allow_projection silently to False when the user sets True explicitly? That wouldn't be ideal

Reply:
Yeah, good question. There is no documentation for this yet, so silently ignoring the `allow_projection=True` default certainly seems strange. The idea is that we **cannot** support column projection unless the `func` argument includes a `"columns"` argument. The purpose of `allow_projection` is to provide the user with an "opt out" mechanism for the case that their `func` argument does include a `"columns"` argument, but they do not want dask-expr to project column selections into their `FromMap` layer.

Maybe there is a better API for this? I get the sense that we want users to get column projection as automatically as possible. I suppose we can remote `allow_projection` as a `from_map` argument if the documentation is clear that `simplify` will always **use**  the `"columns"` argument if `func` has one in its signature?

Reply:
>  I suppose we can remote allow_projection as a from_map argument if the documentation is clear that simplify will always use the "columns" argument if func has one in its signature?

I think I like starting with this and moving to more arguments later on if it becomes necessary

Reply:
Yeah, I agree with this.